### PR TITLE
Add  PrometheusRules to Kubernetes manifests and Openshift templates

### PR DESCRIPTION
**Pull request by metalmatze**
We can add these PrometheusRule files to our OpenShift templates and deploy them like that to our namespace. The Prometheus monitoring those is finally able to pick them up from our namespace now.

/cc @aditya-konarde @squat @kakkoyun

### CODE REVIEW

**Comment by metalmatze** [comment]
Rebased ready for another review.

File: environments/kubernetes/kube-thanos.libsonnet [hunk 1]
```diff
@@ -32,6 +38,47 @@ local jaegerAgent = import '../../components/jaeger-agent.libsonnet';
...
+            thanosReceiveControllerSelector: 'job="%s"' % self.thanosReceiveControllerJobPrefix,
+
+            local config = self,
+            // We build alerts for the presence of all these jobs.
+            jobs: {
+              ThanosQuerier: config.thanosQuerierSelector,
+              ThanosStore: config.thanosStoreSelector,
+              ThanosCompact: config.thanosCompactSelector,
+            } + {
+              ['ThanosReceive' + capitalize(tenant.hashring)]: 'job="thanos-receive-%s", namespace="telemeter-stage"' % tenant.hashring
```
Comment (brancz) [inline environments/kubernetes/kube-thanos.libsonnet]: `namespace="telemeter-stage"`?

Comment (metalmatze) [reply environments/kubernetes/kube-thanos.libsonnet]: Yep. That was overlooked...

File: environments/kubernetes/manifests/thanos-rules-prometheusrule.yaml [hunk 2]
```diff
@@ -138,9 +137,9 @@ spec:
           $value }} seconds for HTTP requests.
       expr: |
         (
-          histogram_quantile(0.99, sum by (job, le) (http_request_duration_seconds_bucket{job=~"thanos-receive.*", namespace="telemeter-stage", handler="receive"})) > 10
+          histogram_quantile(0.99, sum by (job, le) (http_request_duration_seconds_bucket{job=~"thanos-receive-.*", handler="receive"})) > 10
         and
-          sum by (job) (rate(http_request_duration_seconds_count{job=~"thanos-receive.*", namespace="telemeter-stage", handler="receive"}[5m])) > 0
+          sum by (job) (rate(http_request_duration_seconds_count{job=~"thanos-receive-.*", handler="receive"}[5m])) > 0
```
Comment (aditya-konarde) [inline environments/kubernetes/manifests/thanos-rules-prometheusrule.yaml]: Note that this regex match will also select the controllers. Not sure if that's intended.

Comment (aditya-konarde) [reply environments/kubernetes/manifests/thanos-rules-prometheusrule.yaml]: Nvm I totally missed the handler selector.

Comment (metalmatze) [reply environments/kubernetes/manifests/thanos-rules-prometheusrule.yaml]: That would be true if we didn't have the handler="receive". So we should be fine.

File: environments/kubernetes/manifests/thanos-rules-prometheusrule.yaml [hunk 3]
```diff
@@ -177,11 +176,21 @@ spec:
...
         message: Thanos Receive {{$labels.job}} has not been able to reload hashring
           configurations.
-      expr: avg(thanos_receive_config_last_reload_successful{job=~"thanos-receive.*",
-        namespace="telemeter-stage"}) by (job) != 1
+      expr: avg(thanos_receive_config_last_reload_successful{job=~"thanos-receive-.*"})
+        by (job) != 1
       for: 5m
       labels:
         severity: warning
+  - name: thanos-sidecar.rules
```
Comment (kakkoyun) [inline environments/kubernetes/manifests/thanos-rules-prometheusrule.yaml]: We don't have `sidecar` in our setup, I think we should remove this alert.

Comment (metalmatze) [reply environments/kubernetes/manifests/thanos-rules-prometheusrule.yaml]: Good catch. I made sure that the SidecarDown alerts are gone. Will take care of this.

File: environments/kubernetes/kube-thanos.libsonnet [hunk 4]
```diff
@@ -10,6 +10,12 @@ local sa = k.core.v1.serviceAccount;
 local role = k.rbac.v1.role;
 local rolebinding = k.rbac.v1.roleBinding;
 local jaegerAgent = import '../../components/jaeger-agent.libsonnet';
+local thanos = (import 'thanos-mixin/mixin.libsonnet');
```
Comment (squat) [inline environments/kubernetes/kube-thanos.libsonnet]: we don't need parens here, no?

File: environments/kubernetes/kube-thanos.libsonnet [hunk 5]
```diff
@@ -10,6 +10,12 @@ local sa = k.core.v1.serviceAccount;
 local role = k.rbac.v1.role;
 local rolebinding = k.rbac.v1.roleBinding;
 local jaegerAgent = import '../../components/jaeger-agent.libsonnet';
+local thanos = import 'thanos-mixin/mixin.libsonnet';
+local thanosReceiveController = (import 'thanos-receive-controller-mixin/mixin.libsonnet');
```
Comment (squat) [inline environments/kubernetes/kube-thanos.libsonnet]: here too ^

**Comment by metalmatze** [comment]
Both comments addressed. Double checked that there are no rules for components in our files that we don't want.

**Pull request merged**